cvat/apps/engine/serializers.py

```diff
@@ -1980,63 +1982,64 @@ class Meta:
         read_only_fields = ('path',)
 
 
-def _update_related_storages(instance, validated_data):
-    for storage in ('source_storage', 'target_storage'):
-        new_conf = validated_data.pop(storage, None)
+def _update_related_storages(
+    instance: Union[models.Project, models.Task],
+    validated_data: Dict[str, Any],
+) -> None:
+    for storage_type in ('source_storage', 'target_storage'):
+        new_conf = validated_data.pop(storage_type, None)
 
         if not new_conf:
             continue
 
+        cloud_storage = None
         cloud_storage_id = new_conf.get('cloud_storage_id')
-        if cloud_storage_id:
-            _validate_existence_of_cloud_storage(cloud_storage_id)
+        new_location = new_conf.get('location')
 
         # storage_instance maybe None
-        storage_instance = getattr(instance, storage)
-        if not storage_instance:
-            storage_instance = models.Storage(**new_conf)
-            storage_instance.save()
-            setattr(instance, storage, storage_instance)
-            continue
+        storage_instance = getattr(instance, storage_type)
 
-        new_location = new_conf.get('location')
-        storage_instance.location = new_location or storage_instance.location
-        storage_instance.cloud_storage_id = new_conf.get('cloud_storage_id', \
-            storage_instance.cloud_storage_id if not new_location else None)
+        if not cloud_storage_id and new_location == models.Location.CLOUD_STORAGE:
```

Comment:
What about the case where `cloud_storage_id` is `None`, current location is `CLOUD_STORAGE`, and `new_location` is unspecified? Seems like that should be prohibited too.